hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/tasks/FileSizeCountTask.java

```diff
@@ -49,33 +52,28 @@
   private static final Logger LOG =
       LoggerFactory.getLogger(FileSizeCountTask.class);
 
-  private int maxBinSize = -1;
-  private long maxFileSizeUpperBound = 1125899906842624L; // 1 PB
-  private long[] upperBoundCount;
-  private long oneKb = 1024L;
+  // 1125899906842624L = 1PB
+  private static final long MAX_FILE_SIZE_UPPER_BOUND = 1125899906842624L;
   private FileCountBySizeDao fileCountBySizeDao;
+  // Map to store file counts in each <volume,bucket,fileSizeUpperBound>
+  private Map<FileSizeCountKey, Long> fileSizeCountMap;
```

Comment:
Since we don't read everything from the DB at init time, 'fileSizeCountMap' can be a local variable created inside the 'process' and 'reprocess' methods.

Reply:
Moving it to local variable would then require `writeCountsToDB`, `handlePutKeyEvent` and `handleDeleteKeyEvent` methods to take `Map<FileSizeCountKey, Long> fileSizeCountMap` as another argument. Currently, initializing this map in reprocess and process is just enough since both methods will never be called at the same time.

Reply:
IMO, it is OK to have the handleDeleteKeyEvent & handlePutKeyEvent and writeCountsToDB to take the map as a parameter. They are meant to be stateless helper methods that act on the input given to them. In the current implementation, there is an assumption that the task's implementation makes of the higher level caller, which may not be true in the future.